app/services/events/stores/aggregated_clickhouse_store.rb

```diff
@@ -0,0 +1,206 @@
+        end
+      end
+
+      # NOTE: check if an event created before the current on belongs to an active (as in present and not removed)
+      #       unique property
+      def active_unique_property?(event)
+        # TODO(pre-aggregation): Implement
+      end
+
+      def unique_count
+        # TODO(pre-aggregation): Implement
+      end
+
+      def unique_count_breakdown
+        # TODO(pre-aggregation): Implement
+      end
+
+      def prorated_unique_count
+        # TODO(pre-aggregation): Implement
+      end
+
+      def prorated_unique_count_breakdown(with_remove: false)
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_unique_count
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_prorated_unique_count
+        # TODO(pre-aggregation): Implement
+      end
+
+      def max
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_max
+        # TODO(pre-aggregation): Implement
+      end
+
+      def last
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_last
+        # TODO(pre-aggregation): Implement
+      end
+
+      def sum_precise_total_amount_cents
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_sum_precise_total_amount_cents
+        # TODO(pre-aggregation): Implement
+      end
+
+      def sum
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_sum
+        # TODO(pre-aggregation): Implement
+      end
+
+      def prorated_sum(period_duration:, persisted_duration: nil)
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_prorated_sum(period_duration:, persisted_duration: nil)
+        # TODO(pre-aggregation): Implement
+      end
+
+      def sum_date_breakdown
+        # TODO(pre-aggregation): Implement
+      end
+
+      def weighted_sum(initial_value: 0)
+        # TODO(pre-aggregation): Implement
+      end
+
+      def grouped_weighted_sum(initial_values: [])
+        # TODO(pre-aggregation): Implement
+      end
+
+      # NOTE: not used in production, only for debug purpose to check the computed values before aggregation
+      def weighted_sum_breakdown(initial_value: 0)
+        # TODO(pre-aggregation): Implement
+      end
+
+      def aggregated_arel_table
+        @aggregated_arel_table ||= ::Clickhouse::EventsAggregated.arel_table
+      end
+
+      def formated_grouped_by_values
+        value = grouped_by_values
+          .sort_by { |key, value| key }
+          .map { |key, value| "\"#{key}\": \"#{value || NIL_GROUP_VALUE}\"" }
+          .join(",")
+
+        "{#{value}}"
+      end
+
+      # NOTE: returns the values for each groups
+      #       The result format will be an array of hash with the format:
+      #       [{ groups: { 'cloud' => 'aws', 'region' => 'us_east_1' }, value: 12.9 }, ...]
+      def prepare_grouped_result(rows, timestamp: false)
+        rows.map do |row|
```

Comment:
For readability, I'd maybe suggest:

```suggestion
        rows.map do |row|
          group_by_string, value = row
```